crates/hotshot-state-prover/src/circuit.rs

```diff
@@ -116,8 +123,28 @@ where
         let threshold = u256_to_field::<F>(threshold);
         let threshold_var = circuit.create_public_variable(threshold)?;
 
-        // TODO(Chengyu): put in the hotshot state
-        let public_inputs = vec![threshold];
+        let view_number_f = F::from(lightclient_state.view_number as u64);
+        let block_height_f = F::from(lightclient_state.block_height as u64);
+        let lightclient_state_var = [
+            circuit.create_public_variable(view_number_f)?,
+            circuit.create_public_variable(block_height_f)?,
+            circuit.create_public_variable(lightclient_state.block_comm)?,
+            circuit.create_public_variable(lightclient_state.fee_ledger_comm)?,
+            circuit.create_public_variable(lightclient_state.stake_table_comm.0)?,
+            circuit.create_public_variable(lightclient_state.stake_table_comm.1)?,
+            circuit.create_public_variable(lightclient_state.stake_table_comm.2)?,
+        ];
```

Comment:
this variable should be of type `LightClientStateVar` instead of an array, right?

also that struct should have a `LightClientStateVar::new(state) -> Self` to help.

Reply:
This is for later when you need to offer all these vars as a input to Schnorr signature verification. Maybe I should have a struct that has `get_*()` interfaces and implement `AsRef<[Variable]>`.

Reply:
check 599a75d